joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala

```diff
@@ -2293,89 +2116,92 @@ class AstCreator(filename: String, javaParserAst: CompilationUnit, global: Globa
     assignments.toList
   }
 
-  private def completeInitForConstructor(
-    partialConstructor: PartialConstructor,
-    identifier: NewIdentifier,
-    order: Int
-  ): Ast = {
+  private def completeInitForConstructor(partialConstructor: PartialConstructor, identifier: NewIdentifier): Ast = {
     val initNode = partialConstructor.initNode
-      .order(order)
-      .argumentIndex(order)
 
     val objectNode = identifier.copy
-      .order(0)
-      .argumentIndex(0)
 
     val args = partialConstructor.initArgs
 
-    Ast(initNode)
-      .withChild(Ast(objectNode))
-      .withReceiverEdge(initNode, objectNode)
-      .withChildren(args)
-      .withArgEdge(initNode, objectNode)
-      .withArgEdges(initNode, args.flatMap(_.root))
+    astWithRecvArgEdge(initNode, args.toList, Some(Ast(objectNode)))
   }
 
-  def astsForVariableDecl(varDecl: VariableDeclarationExpr, order: Int): Seq[Ast] = {
+  def astsForVariableDecl(varDecl: VariableDeclarationExpr): Seq[Ast] = {
 
-    val locals    = localsForVarDecl(varDecl, order)
+    val locals    = localsForVarDecl(varDecl)
     val localAsts = locals.map { Ast(_) }
 
     locals.foreach { local =>
       scopeStack.addToScope(local.name, local)
     }
 
-    val assignOrder = order + locals.size
     val assignments =
-      assignmentsForVarDecl(varDecl.getVariables.asScala, line(varDecl), column(varDecl), assignOrder)
+      assignmentsForVarDecl(varDecl.getVariables.asScala, line(varDecl), column(varDecl))
 
     localAsts ++ assignments
   }
 
-  def callAst(rootNode: NewCall, args: Seq[Ast]): Ast = {
+  private def setArgumentIndex(arguments: Seq[Ast]): Unit = {
+    withIndex(arguments) { case (a, i) =>
+      a.root.collect { case x: ExpressionNew =>
+        x.argumentIndex = i
+        x.order = i
+      }
+    }
+  }
+
+  // TODO: This is an almost exact copy of the method in x2cpg. Fix this.
+  // The reason this exists here is that some dataflow tests are failing without the arg edge
+  // to the receiver.
```

Comment:
It depends on the language whether the receiver is also an argument to the called method or not. So the x2cpg version of this method should support both. What is its name? I could not find a `astWithRecv...` named method in x2cpg.

Reply:
That's a good point. The method in `x2cpg` is actually called `callAst`. I originally called this something similar, but later changed it to avoid confusing the two without also updating the comment.

Reply:
To address your actual point here, I'll update the `callAst` method in x2cpg as well and switch back to that.